### ISSUE TITLE
Slow generation of lot of pages

### ISSUE THREAD

**Issue by camdav** [jantimon/html-webpack-plugin#713]
Hi, i'm using webpack+html-webpack-plugin to build a static website based on 2 .ejs template that should create a lot of pages.
To do this i have a configuration file like this:

```js
module.exports = {
    entry: {
        vendorsAngular: path.resolve(__dirname, '../www/vendorsAngular.js'),
        vendors: path.resolve(__dirname, '../www/vendors.js'),
        app: path.resolve(__dirname, '../www/main.js'),
        appAngular: path.resolve(__dirname, '../www/mainAngular.js'),
    },
    output: {
        path: path.resolve(__dirname, '../build'),
        filename: '[name].bundle.js'
    },
    resolve: {
        modules: ['../www/lib', '../node_modules'],
        extensions: ['.js'],
        alias: {
            'jquery': 'jquery/src/jquery',
        }
    },
    module: {
        loaders: [
            {
                test: /\.css$/,
                loader: 'style-loader!css-loader'
            },
        ]
    },
    plugins: [
        new HtmlWebpackPlugin({
            title: 'title1',
            template: path.resolve(__dirname, '../www/index.ejs'),
            filename: addPath('index.html'),
            excludeChunks: ['vendorsAngular', 'appAngular']
        }),
        new HtmlWebpackPlugin({
            title: 'page 1',
            template: path.resolve(__dirname, '../www/page.ejs'),
            filename: addPath('page.html'),
            excludeChunks: ['vendorsAngular', 'appAngular']
        }),
    ]
};

var titles = ['title2', 'title3' ... 'title1000'];
var createOtherPages = function() {
    titles.forEach(function(title) {
        module.exports.plugins.unshift(new HtmlWebpackPlugin({
            title: title,
            template: path.resolve(__dirname, '../www/index.ejs'),
            filename: title + '.html',
            excludeChunks: ['vendorsAngular', 'appAngular']
        }));
    });
};
createOtherPages();

```

The two questions are:
Is this the best approach to acheive what i want?
Webpack build is very slow, and fail because there is no memory enought. Do you have any suggestions to improve the performance beside to increase npm memory with --max_old_space_size npm param?

Thanks in advance.
Davide

**Comment by mastilver** [comment]
Uhm, for your use case I would probably use https://github.com/danethurber/webpack-manifest-plugin and another build tool like `gulp` that would read the `manifest.json` to output your desired html files

**Comment by scinos** [comment]
This should get better with #723

**Comment by jantimon** [comment]
Released: 2.30.0
